src/Symfony/Component/HttpKernel/Tests/HttpCache/HttpCacheTest.php

```diff
@@ -809,6 +809,40 @@ public function testValidatesCachedResponsesWithETagAndNoFreshnessInformation()
         $this->assertTraceNotContains('miss');
     }
 
+    public function testServesResponseWhileFreshAndRevalidatesWithLastModifiedInformation()
+    {
+        $time = \DateTime::createFromFormat('U', time());
+
+        $this->setNextResponse(200, array(), 'Hello World', function (Request $request, Response $response) use ($time) {
+            $response->setSharedMaxAge(10);
+            $response->headers->set('Last-Modified', $time->format(DATE_RFC2822));
+        });
+
+        // prime the cache
+        $this->request('GET', '/');
+
+        // next request before s-maxage has expired: Serve from cache
+        // without hitting the backend
+        $this->request('GET', '/');
+        $this->assertHttpKernelIsNotCalled();
+        $this->assertEquals(200, $this->response->getStatusCode());
+        $this->assertEquals('Hello World', $this->response->getContent());
+        $this->assertTraceContains('fresh');
+
+        sleep(15); // expire the cache
+
+        $this->setNextResponse(304, array(), '', function (Request $request, Response $response) use ($time) {
+            $this->assertEquals($time->format(DATE_RFC2822), $request->headers->get('IF_MODIFIED_SINCE'));
```

Comment:
we all do the same mistake, so I don't need the test to fail to spot this kind of errors :)